extensions/datastores/dynamodb/src/main/java/mil/nga/giat/geowave/datastore/dynamodb/mapreduce/GeoWaveDynamoDBRecordReader.java

```diff
@@ -0,0 +1,112 @@
+package mil.nga.giat.geowave.datastore.dynamodb.mapreduce;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+
+import mil.nga.giat.geowave.core.store.CloseableIterator;
+import mil.nga.giat.geowave.core.store.adapter.AdapterStore;
+import mil.nga.giat.geowave.core.store.base.BaseDataStore;
+import mil.nga.giat.geowave.core.store.filter.QueryFilter;
+import mil.nga.giat.geowave.core.store.index.PrimaryIndex;
+import mil.nga.giat.geowave.core.store.query.DistributableQuery;
+import mil.nga.giat.geowave.core.store.query.QueryOptions;
+import mil.nga.giat.geowave.datastore.dynamodb.DynamoDBOperations;
+import mil.nga.giat.geowave.datastore.dynamodb.query.InputFormatDynamoDBRangeQuery;
+import mil.nga.giat.geowave.datastore.dynamodb.split.DynamoDBSplitsProvider;
+import mil.nga.giat.geowave.mapreduce.input.GeoWaveInputKey;
+import mil.nga.giat.geowave.mapreduce.splits.GeoWaveRecordReader;
+import mil.nga.giat.geowave.mapreduce.splits.GeoWaveRowRange;
+
+public class GeoWaveDynamoDBRecordReader<T> extends
+		GeoWaveRecordReader<T>
+{
+
+	private DynamoDBOperations dynamoDBOperations;
+
+	public GeoWaveDynamoDBRecordReader(
+			final DistributableQuery query,
+			final QueryOptions queryOptions,
+			final boolean isOutputWritable,
+			final AdapterStore adapterStore,
+			final BaseDataStore dataStore,
+			final DynamoDBOperations dynamoDBOperations ) {
+		super(
+				query,
+				queryOptions,
+				isOutputWritable,
+				adapterStore,
+				dataStore);
+		this.dynamoDBOperations = dynamoDBOperations;
+	}
+
+	@Override
+	protected CloseableIterator queryRange(
+			PrimaryIndex i,
+			GeoWaveRowRange range,
+			List queryFilters,
+			QueryOptions rangeQueryOptions ) {
+		// this will take the inputsplit
+
+		return new InputFormatDynamoDBRangeQuery(
+				dataStore,
+				adapterStore,
+				dynamoDBOperations,
+				i,
+				DynamoDBSplitsProvider.unwrapRange(range),
+				queryFilters != null ? queryFilters : new ArrayList<QueryFilter>(), // TODO
```

Comment:
Previously, there was no check if queryFilters was null or not, I thought it could be problematic, so I added a check and a TODO to remind myself that i had (turns out the bug was somewhere else, so we probably don't need this check)